src/mongocxx/test/collection.cpp

```diff
@@ -2808,4 +2812,90 @@ TEST_CASE("Ensure that the WriteConcernError 'errInfo' object is propagated", "[
     REQUIRE(contains_err_info);
 }
 
+TEST_CASE("expose writeErrors[].errInfo", "[collection]") {
+
+    // A helper for checking that an error document is well-formed according to our requirements:
+    auto writeErrors_well_formed = [](const bsoncxx::document::view& reply_view) -> bool {
+        if (!reply_view["writeErrors"]) {
+            return false;
+        }
+
+        const auto& errdoc = reply_view["writeErrors"][0];
+
+        auto error_code = errdoc["code"].get_int32();
+
+        // The code should always be 121 (DocumentValidationFailure):
+        if (121 != error_code) {
+            std::ostringstream os;
+            os << "writeErrors expected to have code 121, but had " << error_code << " instead";
+            throw std::runtime_error(os.str());
+        }
+
+        // We require the "details" field be present:
+        if (!errdoc["errInfo"]["details"]) {
+            throw std::runtime_error("no \"details\" field in \"writeErrors\"");
+        }
+
+        return true;
+    };
+
+    // Set up our test environment:
+    instance::current();
+
+    mongocxx::options::apm apm_opts;
+
+    auto client_opts = mongocxx::options::client();
```

Comment:
```suggestion
    auto client_opts = test_util::add_test_server_api();
```

Like the C driver, the C++ driver conditionally tests with the server API version. This may resolve the task failure on [Ubuntu 18.04 Debug Latest Versioned API](https://spruce.mongodb.com/version/621928ed5623437eeb06c925/tasks?page=0&variant=%5Eversioned%20api__os~ubuntu-1804_mongodb_version~latest%24).

Reply:
+1 I didn't see this locally and indeed put the cart before the horse-- apologies.